src/app/share/views/SharedLinksView/SharedView.tsx

```diff
@@ -115,6 +117,49 @@ function SharedView(props: SharedViewProps): JSX.Element {
   const [ownerEncryptionKey, setOwnerEncryptionKey] = useState<null | string>(null);
   const pendingInvitations = useAppSelector((state: RootState) => state.shared.pendingInvitations);
 
+  const onItemDropped = async (item, monitor: DropTargetMonitor) => {
+    const droppedData: any = monitor.getItem();
+
+    const transformedObject = droppedData.files.reduce((acc, file, index) => {
+      acc[index] = file;
+      return acc;
+    }, {});
+
+    await onUploadFileInputChanged({
+      files: {
+        ...transformedObject,
+        length: droppedData.files.length,
+      },
+    });
+  };
+
+  const [{ isOver, canDrop }, drop] = useDrop(
+    () => ({
+      accept: [NativeTypes.FILE],
+      collect: (monitor) => ({
+        isOver: monitor.isOver(),
+        canDrop: monitor.canDrop(),
+      }),
+      canDrop: (item: any, monitor): boolean => {
+        const droppedType = monitor.getItemType();
+        const canDrop = droppedType === NativeTypes.FILE && !!sharedNamePath.length && !isCurrentUserViewer();
```

Comment:
extract `!!sharedNamePath.length` to a variable to a better understanding the reason of the condition.